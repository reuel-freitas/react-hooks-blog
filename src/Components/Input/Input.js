import React from 'react';
import P from 'prop-types';

export const Input = ({ handleChange, searchValue = '' }) => (
  <input
    type="search"
    onChange={handleChange}
    value={searchValue}
    placeholder="Type to search..."
    autoFocus
    style={styles.input}
  />
);

Input.propTypes = {
  handleChange: P.func.isRequired,
  searchValue: P.string.isRequired,
};

const styles = {
  input: {
    marginTop: 20,
    width: 220,
    height: 30,
    padding: 10,
  },
};
