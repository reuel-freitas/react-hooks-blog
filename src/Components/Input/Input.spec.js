import { render, screen } from '@testing-library/react';
// import { userEvent } from '@testing-library/user-event';
import { Input } from './Input';

describe('<Input />', () => {
  it('should have a value of searchValue', () => {
    const fn = jest.fn();
    render(<Input handleChange={fn} searchValue={'Testando'} />);

    const input = screen.getByPlaceholderText(/type to search/i);
    expect(input).toBeInTheDocument();

    expect(input.value).toBe('Testando');
  });

  // it('should call handleChange function on each key pressed', () => {
  //   const fn = jest.fn();
  //   render(<Input handleChange={fn} searchValue="um valor" />);

  //   const input = screen.getByPlaceholderText(/type to search/i);
  //   const value = 'o valor';

  //   userEvent.type(input, value);

  //   expect(input.value).toBe('um valor');
  //   expect(fn).toHaveBeenCalledTimes(value.length);
  // });

  it('should matchs snapshot', () => {
    const fn = jest.fn();
    const { container } = render(<Input handleChange={fn} searchValue="" />);
    expect(container).toMatchSnapshot();
  });
});
