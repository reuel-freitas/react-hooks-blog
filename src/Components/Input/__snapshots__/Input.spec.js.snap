// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should matchs snapshot 1`] = `
<div>
  <input
    placeholder="Type to search..."
    style="margin-top: 20px; width: 220px; height: 30px; padding: 10px;"
    type="search"
    value=""
  />
</div>
`;
